blogposts/2023-02-10-adding-snapshot-testing-to-rnw-apps.md

```diff
@@ -0,0 +1,93 @@
+2. You can also check forum posts from the community on how they mocked particular methods for a given module.
+3. React Native Gallery has a wide set of community modules it has tests for. Check out our [`jest-setup.js`](https://github.com/microsoft/react-native-gallery/blob/main/jest-setup.js) to see how we mocked needed modules. 
+
+Here is React Native Gallery's Jest customization within [`package.json`](https://github.com/microsoft/react-native-gallery/blob/main/package.json):
+```json
+"jest": {
+    "preset": "react-native",
+    "modulePathIgnorePatterns": [
+      "<rootDir>/lib/"
+    ],
+    "transformIgnorePatterns": [
+      "node_modules/(?!(react-native|react-native-xaml|@react-native|react-native-windows|react-native-config|@react-native-community|react-native-print|react-native-webview|react-native-windows-hello|react-native-permissions|react-native-tts|react-native-gesture-handler)/)"
+    ],
+    "setupFiles": [
+      "<rootDir>/jest-setup.js"
+    ]
+  },
+```
+
+Once you have all of your individual tests running and passing, you might see that Jest is still exiting with exit code 1. This occurred in React Native Gallery. After some searching, I discovered it was because some of the components our tests were rendering contained asynchronous calls and updates. This results in Jest throwing errors because it cannot ensure that all tasks such as renders, user events, or data fetching were completed and applied before the snapshot match assertion was made. To resolve this error, you'll need to wrap the render of the component you wish to snapshot in React's `act()` function. Then, following the `act()` call, make your snapshot assertion. 
+
+Here's a code snippet of the test for React Native Gallery's `DateTimePicker` page:
+```js
+test('TimePicker Example Page', async () => {
+  let tree;
+  await act(() => {
+    tree = create(<TimePickerExamplePage />).toJSON();
+  });
+  expect(tree).toMatchSnapshot();
+});
+```
+
+Concluding, in this article I've shared my process for adding Jest Snapshot testing to a React Native Windows application. Adding snapshot testing to React Native Gallery has helped us track visual changes to app components occur and validate that new releases of React Native Windows do not break community modules. Before leaving, I also wanted to share a couple of recommendations:
+1. The code for writing the tests is fairly straightforward; getting Jest configured correctly for your app is where the challenge kicks in. If you're working with an app with a number of community modules, tackle adding tests which use each community module one at a time.
+2. Develop incrementally. Start with a test case which simply renders a `<View/>` control. Make sure your base case runs successfully. Then, test as you go. Jest errors aren't always clear, so the more granular you can make your changes between runs of the tests, the easier you'll be able to diagnose what's wrong.
```

Comment:
Consider putting the "why" before the "what".

```suggestion
2. Develop incrementally. Since Jest errors aren't always clear, consider making granular changes between runs of the tests. That way it'll be easier to diagnose what's wrong. For example, start with a test case that simply renders a `<View/>` component. Make sure your base case runs successfully -- then you can test as you go.
```